parseAndPopulate/groupings.py

```diff
@@ -116,10 +119,11 @@ def _parse_and_load_api(self) -> tuple[int, int]:
         self.logger.debug('Parsing sdo files sent via API')
         with open(os.path.join(self.dir_paths['json'], 'request-data.json'), 'r') as f:
             sdos_json = json.load(f)
-        sdos_list: t.List[dict] = sdos_json['modules']['module']
+        sdos_list: list[Module.AdditionalModuleInfo] = sdos_json['modules']['module']
         sdos_count = len(sdos_list)
         for i, sdo in enumerate(sdos_list, start=1):
             # remove diacritics
+            assert 'source-file' in sdo
```

Comment:
is it okay to raise an exception here? as I can see, we don't use try/except block in the ```parse_directory.py``` script for groupings so it would result in an actual error, maybe we should just use ```continue``` here if 'source-file' not in sdo? and assert isn't the best option if we really need to raise an exception here, I'd suggest using just KeyError here

Reply:
At this point the dict is guaranteed to have a `'source-file'` key, this is checked by the api endpoint that that causes this script to be called (see [here](https://github.com/YangCatalog/backend/blob/3716c97031572f188513e54970493fa364eb2d5b/api/views/userSpecificModuleMaintenance/moduleMaintenance.py#L284)). The assert can not fail here, I inserted it to help the type checker. I should have included that information in the assert though, I'll fix that.